src/Subscription/Subscriber/MetadataSubscriberAccessorRepository.php

```diff
@@ -31,10 +31,11 @@ final class MetadataSubscriberAccessorRepository implements SubscriberAccessorRe
     public function __construct(
         private readonly iterable $subscribers,
         private readonly SubscriberMetadataFactory $metadataFactory = new AttributeSubscriberMetadataFactory(),
-        array $argumentResolvers = [],
+        iterable $argumentResolvers = [],
     ) {
         $this->argumentResolvers = array_merge(
-            $argumentResolvers,
+        // the check for array is required before PHP 8.2
+            is_array($argumentResolvers) ? $argumentResolvers : iterator_to_array($argumentResolvers),
```

Comment:
```suggestion
            is_array($argumentResolvers) ? $argumentResolvers : array_values(iterator_to_array($argumentResolvers)),
```

to gurantee that the resulting array is a list.

Reply:
I put the array_values outside the whole expression, as we cannot trust the incoming array.

Reply:
We can trust the array here due to the type we defined. But it should be fine as is now.